library/std/diagnostics.qs

```diff
@@ -31,4 +31,77 @@ namespace Microsoft.Quantum.Diagnostics {
         }
     }
 
+    /// # Summary
+    /// Given two registers, prepares the maximally entangled state
+    /// between each pair of qubits on the respective registers.
+    /// All qubits must start in the |0⟩ state.
+    ///
+    /// # Input
+    /// ## left
+    /// A qubit array in the |0...0⟩ state
+    /// ## right
+    /// A qubit array in the |0...0⟩ state
+    internal operation PrepareEntangledState (left : Qubit[], right : Qubit[]) : Unit
+    is Adj + Ctl {
+        Fact( Length(left) == Length(right), "PrepareEntangledState: Lengths of qubits registers must be equal.");
+        for idxQubit in 0 .. Length(left) - 1  {
+            H(left[idxQubit]);
+            CNOT(left[idxQubit], right[idxQubit]);
+        }
+    }
+    
+    /// # Summary
+    /// Given two operations, asserts that they act identically for all input states.
```

Comment:
I like the updated name to match the behavior of returning a Boolean (and the fact that it matches `CheckZero` and `CheckAllZero`)! Looks like the doc comment needs updating to remove references to "assert" and "assertion" so that it's consistent.

Reply:
Updated.